src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,177 @@
+            {
+                return {};
+            }
+
+            std::string value = Utility::ConvertToUTF8(GetLocalizedStringFromFont(fontVersion));
+
+            // Version is returned in the format of ex: 'Version 2.137 ;2017'
+            // Extract out the parts between 'Version' and ';'
+            if (Utility::CaseInsensitiveContainsSubstring(value, "Version"))
+            {
+                Utility::FindAndReplace(value, "Version", "");
+            }
+
+            if (Utility::CaseInsensitiveContainsSubstring(value, ";"))
+            {
+                Utility::FindAndReplace(value, ";", "");
+            }
+
+            Utility::Trim(value);
+            std::vector<std::string> items = Utility::Split(value, ' ', true);
+            return Utility::ConvertToUTF16(items[0]);
+        }
+
+        FontFamily GetFontFamilyByIndex(const wil::com_ptr<IDWriteFontCollection>& collection, UINT32 index)
+        {
+            wil::com_ptr<IDWriteFontFamily> family;
+            THROW_IF_FAILED(collection->GetFontFamily(index, family.addressof()));
+            std::wstring familyName = GetFontFamilyName(family);
+
+            std::vector<FontFace> fontFaces;
+            UINT32 fontCount = family->GetFontCount();
+            for (UINT32 fontIndex = 0; fontIndex < fontCount; fontIndex++)
+            {
+                wil::com_ptr<IDWriteFont> font;
+                THROW_IF_FAILED(family->GetFont(fontIndex, font.addressof()));
+
+                wil::com_ptr<IDWriteFontFace> fontFace;
+                THROW_IF_FAILED(font->CreateFontFace(fontFace.addressof()));
+
+                FontFace fontFaceEntry;
+                fontFaceEntry.Name = GetFontFaceName(font);
+                fontFaceEntry.Version = GetFontFaceVersion(font);
+                fontFaceEntry.FilePaths = GetFontFilePaths(fontFace);
+                fontFaces.emplace_back(std::move(fontFaceEntry));
+            }
+
+            FontFamily fontFamily;
+            fontFamily.Name = std::move(familyName);
+            fontFamily.Faces = std::move(fontFaces);
+            return fontFamily;
+        }
+    }
+
+    std::vector<FontFamily> GetInstalledFontFamilies(std::optional<std::wstring> familyName)
+    {
+        wil::com_ptr<IDWriteFactory7> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(factory), factory.put_unknown()));
+
+        wil::com_ptr<IDWriteFontCollection> collection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(collection.addressof(), FALSE));
+
+        std::vector<FontFamily> installedFontFamilies;
+
+        if (familyName.has_value())
+        {
+            UINT32 index;
+            BOOL exists;
+            THROW_IF_FAILED(collection->FindFamilyName(familyName.value().c_str(), &index, &exists));
+
+            if (!exists)
+            {
+                return {};
+            }
+
+            installedFontFamilies.emplace_back(GetFontFamilyByIndex(collection, index));
```

Comment:
```suggestion
            if (exists)
            {
                installedFontFamilies.emplace_back(GetFontFamilyByIndex(collection, index));
            }
```
There are optimizations that are allowed by the compiler when only one thing is returned (can be multiple return statements, but must be the same object on each of them).  It would be more efficient to return `installedFontFamilies` than `{}`.

Additionally, the only thing you are doing after this is the `emplace_back`, so it would also be more efficient for readers to just do that in the conditional block.